forge-game/src/main/java/forge/game/ability/effects/CounterEffect.java

```diff
@@ -253,6 +253,8 @@ private static boolean removeFromStack(final SpellAbility tgtSA, final SpellAbil
 
         params.put(AbilityKey.StackSa, tgtSA);
         params.put(AbilityKey.StackSi, si);
+        params.put(AbilityKey.LastStateBattlefield, game.getLastStateBattlefield());
+        params.put(AbilityKey.LastStateGraveyard, game.getLastStateGraveyard());
```

Comment:
better use a CardZoneTable there?

The Problem lies in Line 55:
https://github.com/Card-Forge/forge/blob/a7d77028e900371769c3a3a199d0a0ff4002bc81/forge-game/src/main/java/forge/game/ability/effects/CounterEffect.java#L55-L56

at this point the SpellAbility doesn't have a LastState so it should fallback to game?

i don't know anymore if AbilityKey.addCardZoneTableParams was the right place for this or if that should have been done in SpellAbilityEffect 
but there are corner cases where you would need the Table, but don't have a SpellAbility

Reply:
ok, I'll try to refine the general usage